### PR TITLE
Adding logging; renaming class

### LINKED ISSUE #7: Logging should be built in
I want a better way to keep track of my LLM-system interactions

### PR DESCRIPTION
**Pull request by benprofitt**
Closes #7 

This should:
- Rename the `Prompt_Formatted` class to `PromptTemplate` (that's what it is now)
- Add in `SystemLogger` class to allow easy logging of activity within a single system
- Change the example code to use the new logging class 
- Other cleanup
- Increment the version number

### CODE REVIEW

File: sdk/python/src/freewayai/llm_system.py [hunk 1]
```diff
@@ -1,7 +1,7 @@
-from .prompt_formatted import Prompt_Formatted
+from .prompt_template import PromptTemplate
 from .database_query_template_format import read_and_format_query
 from .utilities import read_prompt_from_system_and_id
-from .prompt_formatted import Prompt_Formatted
+from .prompt_template import PromptTemplate
```
Comment (michaelcox) [inline sdk/python/src/freewayai/llm_system.py]: We probably didn't catch this before and you did a find/replace but you're importing twice here. Reminds me we should add flake8 and black 😬

File: sdk/python/src/freewayai/system_logger.py [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+    def __init__(self, system : LLMSystem, log_location=None) -> None:
+        self.system = system
+        if log_location is None or isinstance(log_location, str) or hasattr(log_location, 'write'):
+            self.log_location = log_location
+        else:
+            raise ValueError("log_location must be a string or a file pointer or None")
+
+    def log_llm_interaction(self, prompt=None, response=None, prompt_id:str=None, prompt_template:PromptTemplate=None, model:str=None, variables:dict=None):
+        # Format a json body containing each added variable plus the system.system_id
+        log_body = dict()
```
Comment (michaelcox) [inline sdk/python/src/freewayai/system_logger.py:15]: This is all fine but as I'm reading it, it seems like the more common thing to do here would be to have an instance of a logger, namespaced properly, called here. You would pass all of the values to [extra](https://docs.python.org/3/library/logging.html#loggeradapter-objects) in the logger. We could talk about it and update that in a future version though.

**Review by michaelcox — APPROVED** [review]
I think this would be fine to deploy as-is, but left a couple comments below that maybe we want to think though.

File: sdk/python/src/freewayai/system_logger.py (hunk 2 above)
Comment (benprofitt) [reply sdk/python/src/freewayai/system_logger.py:15]: Yup. We can tackle later, agreed

**Pull request merged**